python/lsst/daf/butler/core/_butlerUri.py

```diff
@@ -0,0 +1,1332 @@
+        # Uses os.path.exists so if there is a soft link that points
+        # to a file that no longer exists this will return False
+        return os.path.exists(self.ospath)
+
+    def remove(self) -> None:
+        """Remove the resource."""
+        os.remove(self.ospath)
+
+    def as_local(self) -> Tuple[str, bool]:
+        """Return the local path of the file.
+
+        Returns
+        -------
+        path : `str`
+            The local path to this file.
+        temporary : `bool`
+            Always returns `False` (this is not a temporary file).
+        """
+        return self.ospath, False
+
+    def _force_to_file(self) -> ButlerFileURI:
+        """Force a schemeless URI to a file URI and returns a new URI.
+
+        Returns
+        -------
+        file : `ButlerFileURI`
+            A copy of the URI using file scheme. If already a file scheme
+            the copy will be identical.
+
+        Raises
+        ------
+        ValueError
+            Raised if this URI is schemeless and relative path and so can
+            not be forced to file absolute path without context.
+        """
+        # This is always a file scheme so always return copy
+        return copy.copy(self)
+
+    def relative_to(self, other: ButlerURI) -> Optional[str]:
+        """Return the relative path from this URI to the other URI.
+
+        Parameters
+        ----------
+        other : `ButlerURI`
+            URI to use to calculate the relative path. Must be a parent
+            of this URI.
+
+        Returns
+        -------
+        subpath : `str`
+            The sub path of this URI relative to the supplied other URI.
+            Returns `None` if there is no parent child relationship.
+            Scheme and netloc must match but for file URIs schemeless
+            is also used. If this URI is a relative URI but the other is
+            absolute, it is assumed to be in the parent completely unless it
+            starts with ".." (in which case the path is combined and tested).
+            If both URIs are relative, the relative paths are compared
+            for commonality.
+        """
+        # We know self is a file so check the other. Anything other than
+        # file or schemeless means by definition these have no paths in common
+        if other.scheme and other.scheme != "file":
+            return None
+
+        # for case where both URIs are relative use the normal logic
+        # where a/b/c.txt and a/b/ returns c.txt.
+        if not self.isabs() and not other.isabs():
+            return super().relative_to(other)
+
+        # if we have a relative path convert it to absolute
+        # relative to the supplied parent.  This is solely to handle
+        # the case where the relative path includes ".." but somehow
+        # then goes back inside the directory of the parent
+        if not self.isabs():
+            childUri = other.join(self.path)
+            return childUri.relative_to(other)
```

Comment:
This logic feels weird to me. A relative path is, by convention, relative to the current working directory, but here you're assuming that it's relative to the supplied path, which could produce far different results. Maybe what I'm asking for here is better documentation of what "relative" means in the context of a ButlerURI?

Reply:
For a relative path relative to an absolute path I'm trying to say that the answer is by definition the relative path unless it has `../` in it, in which case it goes up one level. I think that's a fairly well defined definition of relative. The current working directory is not relevant.